src/main.rs

```diff
@@ -382,6 +389,65 @@ impl App {
         }
     }
 
+    fn testcase(&self, args: &ArgMatches) -> Result<()> {
+        let handle = self.current_handle()?;
+        let clash = self.read_clash(&handle)?;
+
+        let mut ostyle = if args.get_flag("no-color") {
+            OutputStyle::plain()
+        } else {
+            OutputStyle::default()
+        };
+        if let Some(show_ws) = args.get_one::<bool>("show-whitespace") {
+            if *show_ws {
+                ostyle.input_whitespace = ostyle.input_whitespace.or(Some(ostyle.input));
+                ostyle.output_whitespace = ostyle.output_whitespace.or(Some(ostyle.output));
+            } else {
+                ostyle.input_whitespace = None;
+                ostyle.output_whitespace = None;
+            }
+        }
+
+        if let Some(values) = args.get_many::<u64>("TESTCASE_INDEX") {
+            let testcases_to_print: Vec<u64> = values.cloned().collect();
+
+            if testcases_to_print.is_empty() {
+                return Err(anyhow!("lul"))
+            }
+
+            // Return an error if any index is out of bounds
+            let num_tests = clash.testcases().len() as u64;
+            for idx in testcases_to_print.iter() {
+                if *idx > num_tests {
+                    return Err(anyhow!("Invalid index {idx}. The clash only has {num_tests} test cases."))
+                }
+            }
+
+            let show_in = args.get_flag("in");
+            let show_out = args.get_flag("out");
+
+            for idx in testcases_to_print {
+                let testcase = &clash.testcases()[(idx - 1) as usize];
+                if !show_in && !show_out {
+                    let styled_title = ostyle.title.paint(format!("#{} {}", idx, testcase.title));
+                    println!("{styled_title}");
+                    println!("{}", ostyle.secondary_title.paint("==== IN ====="));
+                }
+                if show_in || !show_out {
+                    println!("{}", testcase.styled_input(&ostyle));
+                }
+                if !show_in && !show_out {
```

Comment:
Would it not make sense to include `==== OUT ====` even if OUT is the only thing being printed?

Reply:
Alright, fair enough. From the perspective of dumping the output to another program, if they have to find `==== OUT ====` then the `--out` flag is pointless, they could have just parsed the whole output anyway.